### ISSUE TITLE
@next/font font-weight not working correctly in Chrome

### ISSUE THREAD

**Issue by denvradiy** [vercel/next.js#44375]
### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000
    Binaries:
      Node: 14.19.0
      npm: 6.14.16
      Yarn: 1.22.17
      pnpm: 7.13.2
    Relevant packages:
      next: 13.1.1
      eslint-config-next: 13.1.1
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

Font optimization (@next/font)

### Link to the code that reproduces this issue

https://github.com/denvradiy/next-fonts-loading-issue

### To Reproduce

Hard refresh (shift + command+ R) the page multiple times in Chrome in host and font-weight will crash. 

### Describe the Bug

If you refresh the page with reseting the cache (shift + command+ R) multiple times in Chrome then font-weight crashes and displays incorrectly. 

This bug appears only on host (in this case it's vercel). But if I run `next build` and `next start` locally on my computer then it works correctly.

For testing purposes I took generated create-next-app example and didn't change anything

https://github.com/denvradiy/next-fonts-loading-issue - repo to reproduce
https://next-fonts-loading-issue.vercel.app/ - vercel 
https://monosnap.com/file/2uyrSIIo3k4pMpKW0H9RV1JUB4WY9m - screen recording 

### Expected Behavior

Font weight should work correctly in all modern browsers 

### Which browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

**Comment by Evilshoot** [comment]
I have exactly the same issue, would be great to fix it asap

**Comment by Blackglade** [comment]
Experiencing this as well!

**Comment by Archiyopp** [comment]
I have the same problem while using Open Sans Variable font, it only changes between normal and bold weights.

**Comment by nattwasm** [comment]
Also facing this issue

**Comment by maccman** [comment]
Am experiencing this issue too.

**Comment by friconelli** [comment]
I also have the same problem on Chrome, while on Firefox and Safari it works fine

**Comment by tiagobbraga** [comment]
The same problem.

**Comment by tbergeron** [comment]
I think I'm having the same I've been wondering why I can't use anything beyond 700 🤔
I'm trying to figure out a workaround, anyone?

**Comment by nattwasm** [comment]
> I think I'm having the same I've been wondering why I can't use anything beyond 700 🤔 I'm trying to figure out a workaround, anyone?

Maybe try not having a font weight. That fixed it for me

**Comment by tbergeron** [comment]
I ended up working around the bug by using a <style> tag instead:

```
      <style jsx global>{`
        html {
          font-family: ${inter.style.fontFamily};
        }
      `}</style>
```

Now font-weights are working again.

**Comment by denvradiy** [comment]
adding display: 'swap' to the font config fixed my issue

**Comment by elie222** [comment]
Seeing the same issues with Lexend and Inter. Tried a bunch of fixes but nothing working.

**Comment by niraj-khatiwada** [comment]
I'm experiencing the same issue. The font-weight works for the fonts imported from "next/font/google" but it does not work for local fonts at all. Any quick fix?

**Comment by Vincenteliezer** [comment]
experiencing this as well, nextjs 15, i have tried everything!

**Comment by axai-kaizoku** [comment]
I've also tried many things. Spent so many hours fixing this.  But lastly this is working

body {
  -webkit-font-smoothing: antialiased; /* Improves clarity */
  -moz-osx-font-smoothing: grayscale; /* Improves clarity */
}

Add this to your global.css file.

**Comment by timneutkens** [comment]
This is closed as fixed. The behavior you hit came from the `@next/font` default of `font-display: optional` in `13.1.1`: on a hard refresh with an empty cache the font download loses the race, the browser discards it for that page load, and you are left with the fallback, which only has normal and bold. That matches your own finding that `display: 'swap'` resolved it, and it explains why `next build` plus `next start` looked fine locally while the deployed page did not. Waiting on a font you already downloaded is a frustrating thing to debug.

What was checked:

- Rebuilt the reported setup on `next@13.1.1` with `@next/font@13.1.1`, since `denvradiy/next-fonts-loading-issue` now returns 404, and measured rendered text width per `font-weight` in Chromium with a cold cache and the font responses delayed 3s. Weights `400` through `900` collapsed to two widths, fallback only.
- Ran the same measurement on `next@16.3.1-canary.25`. With the current default of `swap`, real graded `Inter` renders on the same cold-cache load. Setting `display: 'optional'` explicitly still shows the old fallback-only result, so the mechanism is intact but is no longer what you get by default.
- The default changed in #45430 and has not been reverted. Current docs list `swap` as the default for `next/font`.

One related thing worth calling out for others in this thread who see weights not applying on a warm cache, which is a different cause: passing a fixed `weight` to a Google font that has a variable `wght` axis makes Google Fonts return a single `@font-face` pinned to that value, so CSS `font-weight` has no effect afterwards. On both `13.1.1` and current canary, `Inter({ weight: '400' })` renders identically at every weight from `400` to `900`, and `Inter({ weight: ['400', '600', '900'] })` snaps `500` to `400` and `700` and `800` to `900`. Omit `weight` to keep the full variable range. If that is your symptom, please open a new issue with a minimal reproduction rather than commenting here.

`13.x` is out of the support window, so upgrade to `16` or `15` if you are still on it.
